harness/determined/common/experimental/trial.py

```diff
@@ -353,15 +350,24 @@ def key(ckpt: checkpoint.Checkpoint) -> Any:
     def __repr__(self) -> str:
         return "Trial(id={})".format(self.id)
 
+    def stream_metrics(self, metric_type: util.MetricType) -> Iterable[TrialMetrics]:
```

Comment:
should this type be just a string? my impression is that `util.MetricType` is for legacy key names? do we even need a separate enum type for it if this is the case?

Reply:
hmm it is not an enum. it's a hint and just an alias to str under the hood IMO this makes it more clear what the argument is about and lets us track the usage of MetricType throughout our codebase by looking at the references of it. you should still be able to pass it raw `str` with no issues.

Reply:
ah, i thought it was an enum type, my bad. your reasoning about making it easier for us to track in the codebase makes sense, but i'm still hesitant about making it a user-facing type alias. my preference is still for the simple "str" type because it makes it explicit to users what inputs to expect. 

having a type alias here also makes me think there are a limited discrete set of types we support (when any generic string is valid) and i have to go look into the code to figure out what `MetricType` is.

Reply:
@hamidzr invited me to comment on this.

I am not opposed to all type aliases, but IMO aliasing `str` like this is not very helpful to end users.  If I were a user, I'd start thinking, "I want to call `stream_metrics`" and I'd look at the docs and say, "what the heck is a util.MetricType" and I'd be annoyed if I dug far enough to find out it's just a `str`.

> makes it more clear what the argument is about

I think for parameters especially (since they are inputs provided by the user), the name is what should make the parameter's purpose clear, and the type hint should make it clear what type is expected.  Everybody knows what a `str` is, nobody knows what a `MetricType` is.

Regarding the name, I'd prefer we name this parameter `series` or something.  "Training" isn't a different "type" of metric, it's a different "series" of metrics.  A user could validly think to themselves, "well I want float-type metrics" and be confused as to how to configure `metric_type` accordingly.

> lets us track the usage of MetricType throughout our codebase

That presumes that people consistently use MetricType, even though mypy won't enforce that to be true.  I sympathize with this desire, but I think canonical names for a variable are the best solution for it.  If everything you might assign a `MetricType` to is named `metric_type`, and you can grep for that name.  We're pretty good about keeping canonical names in the harness, I think, such as with symbols like `self.context`, `self.distributed`, or `core_context`.

Reply:
thanks for the comments! not forcing the user to look up what `MetricType` makes sense especially if they not using an ide that shows it right there eg in a notebook.